lib/msf/core/exploit/http/client.rb

```diff
@@ -320,22 +322,48 @@ def send_request_raw(opts={}, timeout = 20, disconnect = false)
       r = c.request_raw(opts)
 
       if datastore['HttpTrace']
-        print_line('#' * 20)
-        print_line('# Request:')
-        print_line('#' * 20)
-        print_line(r.to_s)
+        request_color, response_color =
+          datastore['HttpTraceColors'].split('/').map { |color| "%#{color}" }
+
+        if datastore['HttpTraceHeadersOnly']
+          request = r.to_s(headers_only: true)
+        else
+          request = r.to_s
+        end
+
+        case datastore['HttpTrace']
+        when 'true'
+          print_line('#' * 20)
+          print_line('# Request:')
+          print_line('#' * 20)
+          print_line(request)
+        when 'color'
+          print_line("%clr%bld#{request_color}#{request}%clr")
+        end
       end
 
       res = c.send_recv(r, actual_timeout)
 
       if datastore['HttpTrace']
-        print_line('#' * 20)
-        print_line('# Response:')
-        print_line('#' * 20)
-        if res.nil?
-          print_line("No response received")
-        else
-          print_line(res.to_terminal_output)
+        response =
+          if res
+            if datastore['HttpTraceHeadersOnly']
+              res.to_terminal_output(headers_only: true)
+            else
+              res.to_terminal_output
+            end
+          else
+            'No response received'
```

Comment:
I think I don't want to colorize this, after all. It's from the server but not _from_ the server.